File: src/Polls/LJPollQuestion.m
//
//  LJQuestion.m
//  PollTest
//
//  Created by Fraser Speirs on Fri Mar 14 2003.
//  Copyright (c) 2003 Connected Flow. All rights reserved.
//

#import "LJPollQuestion.h"


@implementation LJPollQuestion
@synthesize question = theQuestion;

- (NSString *)htmlRepresentation
{
    return @"";
}

#pragma mark Memento Pattern
- (NSDictionary*)memento { return nil; }
- (void)restoreFromMemento: (NSDictionary*)memento {}

#pragma mark NSCoding
- (void)encodeWithCoder:(NSCoder *)encoder
{
    if ([encoder allowsKeyedCoding]) {
        [encoder encodeObject: [self question] forKey:kLJPollQuestionKey];
    } else {
        [NSException raise:NSInvalidArgumentException format:@"LJKit requires keyed coding."];
    }
}

- (instancetype)initWithCoder:(NSCoder *)decoder
{
    self = [super init];
    if (self) {
        [self setQuestion: [decoder decodeObjectForKey:kLJPollQuestionKey]];
    }
    return self;
}

@end

File: src/Polls/LJPollQuestion.swift
//
//  LJPollQuestion.swift
//  Xjournal
//
//  Created by C.W. Betts on 1/23/15.
//
//

import Foundation

let kLJPollQuestionKey = "LJPollQuestion"


class PollQuestion: NSObject, NSSecureCoding {
	
	/// The question name
	var question: String
	
	override init() {
		question = "New Question"
		super.init()
	}
	
	var memento: [String: Any] {
		return [String: Any]()
	}
	
	@objc(restoreFromMemento:) func restore(fromMemento amemento: [String: Any]) {
		
	}
	
	/// Get the HTML representation
	var htmlRepresentation: String {
		return ""
	}
	
	// MARK: NSCoding
	final class var supportsSecureCoding: Bool {
		return true
	}
	
	required init?(coder aDecoder: NSCoder) {
		if let aQues = aDecoder.decodeObject(forKey: kLJPollQuestionKey) as? String {
			question = aQues
			super.init()
		} else {
			return nil
		}
	}
	
	func encode(with aCoder: NSCoder) {
		aCoder.encode(question, forKey: kLJPollQuestionKey)
	}
}

extension PollQuestion {
	
	@available(*, unavailable, renamed: "restore(fromMemento:)")
	@nonobjc func restoreFromMemento(_ amemento: [String: AnyObject]) {
	
	}
}
